playlist2links

```diff
@@ -17,33 +22,44 @@ if [[ -n $(type -p wget) ]];
 then
 	COMMAND="wget -o /dev/null -O '${TMPFILE}.html' '$URL'"
 fi
-if [[ -z "$COMMAND" ]] && [[ -n $(type -p curl) ]];
+if [ -z "$COMMAND" ] && [[ -n $(type -p curl) ]];
 then
 	COMMAND="curl -s -o '${TMPFILE}.html' '$URL'"
 fi
-if [[ -z "$COMMAND" ]];
+if [ -z "$COMMAND" ];
 then
 	echo "Please, install wget or curl to use this script" 1>&2
 	exit 1
 fi
 
-eval $COMMAND
-grep -e "^<tr.*<a href=\"/watch?v=.*$1" "${TMPFILE}.html" | sed 's/.*watch/https:\/\/www.youtube.com\/watch/g' | sed 's/&amp.*//g' > playlist_$1.txt
+eval "$COMMAND"
+
+if $withnames;
+then
+	grep -e "^<tr.*<a href=\"/watch?v=.*$1" "${TMPFILE}.html" |sed 's/.*data-title=\"/\n/g'|sed 's/\"/\n/'|sed 's/.*watch/https:\/\/www.youtube.com\/watch/g'|sed 's/&amp.*//g' > "playlist_$1.txt"
+else
+	grep -e "^<tr.*<a href=\"/watch?v=.*$1" "${TMPFILE}.html" | sed 's/.*watch/https:\/\/www.youtube.com\/watch/g' | sed 's/&amp.*//g' > "playlist_$1.txt"
+fi
 
-while [[ $(grep "continuation=" ${TMPFILE}.html) ]]
-do
-	sed -i 's/_continuation=1/\n/g' ${TMPFILE}.html
-	CONTINUATION=$((sed 's/\(.*\)continuation=/https:\/\/www.youtube.com\/browse_ajax?action_continuation=1&/g' | sed 's/".*//g' | sed 's/\\u0026/\&/g' | tr -d '\\' ) <<< $(grep "continuation=" ${TMPFILE}.html))
-	
+if grep -q "continuation=" "${TMPFILE}.html"
```

Comment:
This needed to be a loop since it's needed an http request every 100 videos [fixed in next commit]